### PR TITLE
Qute: add SectionResolutionContext.evaluate()

### PR DESCRIPTION
**Pull request by mkouba**
https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/Qute.20section.20handler

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]

**Review by FroMage — APPROVED** [review]
This is great, but you should add it to https://quarkus.io/guides/qute-reference#engine-customization which is how I got into this mess :)

**Comment by mkouba** [comment]
@gsmet FYI the `io.quarkus.gradle.devmode.QuarkusDevDependencyDevModeTest` seems to be flaky.

File: docs/src/main/asciidoc/qute-reference.adoc [hunk 1]
```diff
@@ -1538,20 +1538,36 @@ public class CustomSectionFactory implements SectionHelperFactory<CustomSectionF
...
+        return outerScope;
+    }
+    
 
     @Override
     public CustomSectionHelper initialize(SectionInitContext context) {
-        if (context.getParameter("foo") == null) {
-            throw new IllegalStateException("Foo param not found"); <3>
-        }
         return new CustomSectionHelper();
```
Comment (FroMage) [inline docs/src/main/asciidoc/qute-reference.adoc:1557]: This constructor is missing its parameter :(

Comment (FroMage) [reply docs/src/main/asciidoc/qute-reference.adoc:1557]: Not so great, sorry to have taken so long to review :(

**Pull request merged**